src/sentry/integrations/slack/utils/notifications.py

```diff
@@ -111,40 +115,74 @@ def send_incident_alert_notification(
         if incident.status == IncidentStatus.CRITICAL.value:
             payload["reply_broadcast"] = True
 
-    client = SlackClient(integration_id=integration.id)
     success = False
-    try:
-        response = client.post("/chat.postMessage", data=payload, timeout=5)
-        # response should include a "ts" key that represents the unique identifier for the message
-        # referenced at https://api.slack.com/methods/chat.postMessage#examples
-    except ApiError as e:
-        # Record the error code and details from the exception
-        new_notification_message_object.error_code = e.code
-        new_notification_message_object.error_details = {
-            "url": e.url,
-            "host": e.host,
-            "path": e.path,
-            "data": e.json if e.json else e.text,
-        }
-        logger.info("rule.fail.slack_post", exc_info=True)
+    if features.has("organizations:slack-sdk-metric-alert", organization):
+        try:
+            sdk_client = SlackSdkClient(integration_id=integration.id)
+            sdk_response = sdk_client.chat_postMessage(
+                blocks=json_blocks,
+                text=text,
+                channel=str(channel),
+                thread_ts=thread_ts,
+                reply_broadcast=reply_broadcast,
+            )
```

Comment:
we should add the `sdk_response.validate()` call here: https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/web/slack_response.py#L186

Reply:
shouldn't we do this in the client, when we wrap all the functions?

Reply:
It depends where we want to do error handling, and how we want to do error raising. Let's take a look at where it would fit the best

Reply:
actually, it already calls `validate` in the base client https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/web/base_client.py#L311-L319